torch/csrc/profiler/collection.cpp

```diff
@@ -21,6 +21,8 @@ void InputOutputEncoder::push(c10::ArrayRef<const c10::IValue> values) {
       push(value.toTensor());
     } else if (value.isScalar()) {
       tags_.emplace_back(Tag::Scalar);
+      // Scalars are small enough to store as IValues
+      ivalues_.emplace_back(value);
```

Comment:
How does the perf of holding an IValue compare to storing the Scalar directly? In theory the latter should be cheaper, but if its not meaningfully different obviously IValue has a bunch of advantages.

Reply:
For trivially copyable datatypes (which includes all scalars), the only difference in performance between an `IValue` and a `Scalar` would be an extra check on the tag to see if it is trivially copyable, which should not be meaningfully different. 

Both of them are tagged unions.
https://github.com/pytorch/pytorch/blob/fe6aa0844466e5dd2669092eac5edde153108b28/aten/src/ATen/core/ivalue.h#L1124